contracts/CollateralFundV1.sol

```diff
@@ -2,131 +2,155 @@
-            _amount,
-            collateralFundedByAddress[msg.sender]
-        );
-        payable(msg.sender).transfer(_amount);
+    //slither-disable-next-line naming-convention
+    function withdraw(uint256 _amount) external {
+        require(_amount > 0, "Amount should be greator than zero");
+        require(addressToFunderStatus[msg.sender], "The user has not deposited any collateral to the fund");
+        require(_amount <= addressToCollateralAmount[msg.sender], "The user cannot withdraw more than what he deposited");
+
+        // Send the USDC from the fund to the user's wallet and update the user collateral amount.
+        usdcContract.transfer(msg.sender, _amount);
+        addressToCollateralAmount[msg.sender] -= _amount;
+        emit Deposit(msg.sender, _amount, addressToCollateralAmount[msg.sender]);
     }
 
+    /**************************************** Getters ****************************************/
+
     /**
-     * @notice Get the list of users who have funded the smart contract.
-     * @return _ the list of funders
+     * @notice Get the amount of collateral deposited by a user.
+     * @param _userAddress the address of the user.
+     * @return _ the amount deposited by the user.
+     */
+    //slither-disable-next-line naming-convention
+    function getUserCollateralAmount(address _userAddress) external view returns (uint256) {
+        return addressToCollateralAmount[_userAddress];
+    }
+
+    /**
+     * @notice Get the list of users who have deposited collateral to the fund.
+     * @return _ the list of funders.
      */
     function getFunders() external view returns (address[] memory) {
         return funders;
     }
 
+    /**************************************** Setters ****************************************/
+
     /**
-     * @notice Get the amount deposited by a user.
-     * @param _addr address
-     * @return _ amount deposited by a user
+     * @notice Update the amount of collateral deposited by a user.
+     * @param _userAddress the address of the user.
+     * @param _newCollateralAmount the new amount of collateral of the user.
+     * @dev This method is used by the Hub to liquidate users positions.
      */
     //slither-disable-next-line naming-convention
-    function getCollateralFundedByAddress(address _addr)
-        external
-        view
-        returns (uint256)
-    {
-        return collateralFundedByAddress[_addr];
-    }
+    function setUserCollateralAmount(address _userAddress, uint256 _newCollateralAmount) external {
+        require(msg.sender == hubAddress, "Only the hub can call this method");
+        require(_userAddress != address(0), "The address parameter cannot be null");
 
-    function setCollateralFundedByAddress(address _user, uint256 _amount)
-        public
-        onlyHub
-    {
-        collateralFundedByAddress[_user] = _amount;
+        emit CollateralAmountUpdated(_userAddress, addressToCollateralAmount[_userAddress], _newCollateralAmount);
```

Comment:
event always at the end of a function, no?

Reply:
Yes it would be better but I doesn't change anything at the function level. I did this to send an event that shows the previous collateral amount and the new collateral amount.